### PR TITLE
Sanitize data on save [MAILPOET-4942]

### PR DESCRIPTION
**Pull request by samnajian**
[MAILPOET-4942]

[MAILPOET-4942]: https://mailpoet.atlassian.net/browse/MAILPOET-4942?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/lib/API/MP/v1/Segments.php [hunk 1]
```diff
@@ -42,7 +42,9 @@ public function addList(array $data): array {
     $this->validateSegmentName($data);
 
     try {
-      $segment = $this->segmentsRepository->createOrUpdate($data['name'], $data['description'] ?? '');
+      $name = sanitize_text_field($data['name']);
```
Comment (xknown) [inline mailpoet/lib/API/MP/v1/Segments.php:45]: `sanitize_text_field` is just a wrapper around `strip_tags` which might not do what you might expect. For example:
```
> return [sanitize_text_field('hola < aa> "'), sanitize_text_field('hola > aa "')]
array (
  0 => 'hola < aa> "',
  1 => 'hola > aa "',
)
```
If this behavior is fine, please feel free to ignore this comment :)

**Review by costasovo — APPROVED** [review]
Good job!

File: mailpoet/lib/API/MP/v1/Segments.php (hunk 1 above)
Comment (costasovo) [reply mailpoet/lib/API/MP/v1/Segments.php:45]: We still need to add esc_attr when printing the value on the output, so I guess this is fine cc @samnajian.

**Review by Aschepikov — APPROVED** [review]
GJ!

**Pull request merged**